controllers/nutanixcluster_controller.go

```diff
@@ -360,9 +362,8 @@ func (r *NutanixClusterReconciler) reconcileCredentialRefDelete(ctx context.Cont
 
 	if secret.DeletionTimestamp.IsZero() {
 		log.Info(fmt.Sprintf("removing secret %s in namespace %s for cluster %s", secret.Name, secret.Namespace, nutanixCluster.Name))
-		if err := r.Client.Delete(ctx, secret); err != nil {
-			return err
-		}
+		// If the secret is already deleted, don't return error
+		r.Client.Delete(ctx, secret)
```

Comment:
This should check the error is a not found error:

```suggestion
		if err := r.Client.Delete(ctx, secret); err != nil && !errors.IsNotFound(err) {
			return err
		}
```

Reply:
As a follow-up, why can't this be handled by ownership and GC? Is this code even needed?

Reply:
The secret created with separate manifest and set ownership during reconciliation.
Also we planned to refactor these part to improve testability. Right know there is limited ability to test race conditions. I think it's better to improve in separate PR.

Reply:
OK, but let's at least add the improved error checking in this PR.

Reply:
Let's at least introduce the improved error check in this PR.

Reply:
Fixed